### PR TITLE
DBZ-7461 Update doc for snapshot modes alignment

### PR DESCRIPTION
**Pull request by mfvitale**
docs for https://github.com/debezium/debezium/pull/5245

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 1]
```diff
@@ -630,6 +630,56 @@ a|Record the successful completion of the snapshot in the connector offsets.
 
 |===
 
+[id="oracle-connector-snapshot-mode-options"]
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:633]:
```suggestion
[id="mysql-connector-snapshot-mode-options"]
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 2]
```diff
@@ -3347,32 +3403,74 @@ For example, if you set `max.queue.size=1000`, and `max.queue.size.in.bytes=5000
...
+| When `snapshot.locking.mode` is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotLock' interface.
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
+endif::community[]
+
+|[[mysql-property-snapshot-query-mode]]<<mysql-property-snapshot-query-mode, `+snapshot.query.mode+`>>
+|`select_all`
+|Specifies how the connector queries data while performing a snapshot. +
+Set one of the following options:
+
+`select_all`:: The connector performs a `select all` query.
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3460]: I'm concerned about the phrasing of "select all".  I'm worried that users may take that to imply "SELECT *" and that isn't the case when a user configures a column include/exclude configuration.  Maybe we could augment the description of this option like:

```
`select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 3]
```diff
@@ -3124,6 +3157,37 @@ You can set the following values:
...
+| When `snapshot.locking.mode` is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotLock' interface.
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
+endif::community[]
+
+|[[oracle-property-snapshot-query-mode]]<<oracle-property-snapshot-query-mode, `+snapshot.query.mode+`>>
+|`select_all`
+|Specifies how the connector queries data while performing a snapshot. +
+Set one of the following options:
+
+`select_all`:: The connector performs a `select all` query.
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3176]: Same as above.

**Review by Naros — COMMENTED** [review]
Overall LGTM, just a few nit-picks.

If I recall, didn't we decide that there would be some snapshot mode that would allow the user to toggle the various true/false bits of the `Snapshotter` via configuration?  Did we decide that would be follow-up?

**Comment by mfvitale** [comment]
> Overall LGTM, just a few nit-picks.
> 
> If I recall, didn't we decide that there would be some snapshot mode that would allow the user to toggle the various true/false bits of the `Snapshotter` via configuration? Did we decide that would be a follow-up?

Yes, we have this follow-up https://issues.redhat.com/browse/DBZ-7497

File: documentation/modules/ROOT/pages/connectors/mysql.adoc (hunk 2 above)
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:3460]: Totally agreed.

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 4]
```diff
@@ -3173,7 +3173,7 @@ endif::community[]
 |Specifies how the connector queries data while performing a snapshot. +
 Set one of the following options:
 
-`select_all`:: The connector performs a `select all` query.
+`select_all`:: `select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3176]:
```suggestion
`select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.
```
Same?

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 5]
```diff
@@ -3327,7 +3327,7 @@ endif::community[]
 |Specifies how the connector queries data while performing a snapshot. +
 Set one of the following options:
 
-`select_all`:: The connector performs a `select all` query.
+`select_all`:: `select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3330]:
```suggestion
`select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.
```
Same?

**Review by Naros — APPROVED** [review]
LGTM, deferring to @roldanbob's final call.

**Comment by rk3rn3r** [comment]
Look like the commits link the wrong Jira issue? https://issues.redhat.com/browse/DBZ-7462 is handling something else.

**Review by rk3rn3r — CHANGES_REQUESTED** [review]
Looks like the commits link the wrong Jira issue? https://issues.redhat.com/browse/DBZ-7462 is handling something else.
The correct issue is imo [DBZ-7461](https://issues.redhat.com/browse/DBZ-7461).

**Comment by mfvitale** [comment]
@rk3rn3r commits seems fine, only the PR title was wrong. Fixed.

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 6]
```diff
@@ -3100,17 +3123,27 @@ If the snapshot completes successfully, upon next connector start snapshot is no
...
+
+`recovery`:: This is a recovery setting for a connector that has already been capturing changes.
 When you restart the connector, this setting enables recovery of a corrupted or lost database schema history topic.
 You might set it periodically to "clean up" a database schema history topic that has been growing unexpectedly.
 Database schema history topics require infinite retention.
 Note this mode is only safe to be used when it is guaranteed that no schema changes happened since the point in time the connector was shut down before and the point in time the snapshot is taken.
 
 After the snapshot is complete, the connector continues to read change events from the database's redo logs except when `snapshot.mode` is configured as `initial_only`.
 
+`when_needed`:: The connector performs a snapshot upon startup whenever it deems it necessary. That is, when no offsets are available, or when a previously recorded offset specifies a log position that is not available in the server.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3141]:
```suggestion
`when_needed`:: After the connector starts, it performs a snapshot only if it detects one of the following circumstances:
 
* It cannot detect any topic offsets.
* A previously recorded offset specifies a log position that is not available on the server.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 7]
```diff
@@ -3124,6 +3157,37 @@ You can set the following values:
...
 
+ifdef::community[]
+`custom`:: The connector performs a snapshot according to the implementation specified by the xref:oracle-property-snapshot-locking-mode-custom-name[`snapshot.locking.mode.custom.name`] property, which is a custom implementation of the `io.debezium.spi.snapshot.SnapshotLock` interface.
+endif::community[]
+
+ifdef::community[]
+|[[oracle-property-snapshot-locking-mode-custom-name]]<<oracle-property-snapshot-locking-mode-custom-name, `+snapshot.locking.mode.custom.name+`>>
+|No default
+| When `snapshot.locking.mode` is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotLock' interface.
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3168]: Ditto.

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 8]
```diff
@@ -3124,6 +3157,37 @@ You can set the following values:
...
+ifdef::community[]
+`custom`:: The connector performs a snapshot query according to the implementation specified by the xref:oracle-property-snapshot-snapshot-query-mode-custom-name[`snapshot.query.mode.custom.name`] property, which defines a custom implementation of the `io.debezium.spi.snapshot.SnapshotQuery` interface. +
+endif::community[]
+
+This setting enables you to manage snapshot content in a more flexible manner compared to using the xref:oracle-property-snapshot-select-statement-overrides[`snapshot.select.statement.overrides`] property.
+
+ifdef::community[]
+|[[oracle-property-snapshot-snapshot-query-mode-custom-name]]<<oracle-property-snapshot-query-mode-custom-name, `+snapshot.query.mode.custom.name+`>>
+|No default
+| When xref:oracle-property-snapshot-query-mode[`snapshot.query.mode`] is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotQuery' interface.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3187]:
```suggestion
| When xref:oracle-property-snapshot-query-mode[`snapshot.query.mode`] is set to `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotQuery' interface.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 9]
```diff
@@ -3124,6 +3157,37 @@ You can set the following values:
...
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3188]: Ditto.

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 10]
```diff
@@ -171,23 +171,26 @@ If the connector fails, is rebalanced, or stops after Step 1 begins but before S
...
 |`initial` (default)
 |The connector performs a database snapshot when no Kafka offsets topic exists. After the database snapshot completes the Kafka offsets topic is written. If there is a previously stored LSN in the Kafka offsets topic, the connector continues streaming changes from that position.
 
 |`initial_only`
 |The connector performs a database snapshot and stops before streaming any change event records. If the connector had started but did not complete a snapshot before stopping, the connector restarts the snapshot process and stops when the snapshot completes.
 
-|`exported`
-|Deprecated, all modes are lockless.
+|`no_data`
+|The connector never performs snapshots. When a connector is configured this way, its behavior when it starts is as follows. If there is a previously stored LSN in the Kafka offsets topic, the connector continues streaming changes from that position. If no LSN has been stored, the connector starts streaming changes from the point in time when the PostgreSQL logical replication slot was created on the server. This snapshot mode is useful only when you know all data of interest is still reflected in the WAL.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:181]:
```suggestion
|The connector never performs snapshots. 
When a connector is configured this way, after it starts, it behaves as follows:

If there is a previously stored LSN in the Kafka offsets topic, the connector continues streaming changes from that position. 
If no LSN is stored, the connector starts streaming changes from the point at which the PostgreSQL logical replication slot was created on the server. 
Use this snapshot mode only when you know that all data of interest is still reflected in the WAL.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 11]
```diff
@@ -171,23 +171,26 @@ If the connector fails, is rebalanced, or stops after Step 1 begins but before S
...
+
+|[.line-through]`never`
+|Deprecated, see `no_data`
+
+|`when_needed`
+|The connector performs a snapshot upon startup whenever it deems it necessary. That is, when no offsets are available, or when a previously recorded offset specifies a log position that is not available in the server.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:187]:
```suggestion
|After the connector starts, it performs a snapshot only if it detects one of the following circumstances:
 
* It cannot detect any topic offsets.
* A previously recorded offset specifies a log position that is not available on the server.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 12]
```diff
@@ -171,23 +171,26 @@ If the connector fails, is rebalanced, or stops after Step 1 begins but before S
...
 
 ifdef::community[]
 |`custom`
 |The `custom` snapshot mode lets you inject your own implementation of the `io.debezium.spi.snapshot.Snapshotter` interface.
 Set the `snapshot.mode.custom.name` configuration property to the name provided by the `name()` method of your implementation, as specified on the classpath of your Kafka Connect cluster, or included in the connector JAR file, if using the `EmbeddedEngine`.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:192]:
```suggestion
Set the `snapshot.mode.custom.name` configuration property to the name provided by the `name()` method of your implementation. 
The name is specified on the classpath of your Kafka Connect cluster.
If you use the `EmbeddedEngine`, the name is included in the connector JAR file.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 13]
```diff
@@ -171,23 +171,26 @@ If the connector fails, is rebalanced, or stops after Step 1 begins but before S
...
-For more information, see xref:postgresql-custom-snapshot[custom snapshotter SPI].
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:193]: Ditto.

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 14]
```diff
@@ -241,163 +244,9 @@ include::{partialsdir}/modules/all-connectors/proc-stopping-an-incremental-snaps
 include::{partialsdir}/modules/all-connectors/proc-stopping-an-incremental-snapshot-kafka.adoc[leveloffset=+1]
 
 ifdef::community[]
-[[postgresql-custom-snapshot]]
+[[connector-custom-snapshot]]
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:247]: Ditto.

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 15]
```diff
@@ -3418,31 +3267,33 @@ For example, +
...
-`exported` -  deprecated +
- +
+
+`always`:: The connector performs a snapshot each time the connector starts.
+
+`initial`:: The connector performs a snapshot only when no offsets have been recorded for the logical server name.
+
+`initial_only`:: The connector performs an initial snapshot and then stops, without processing any subsequent changes.
+
+`no_data`:: The connector never performs snapshots. When a connector is configured this way, its behavior when it starts is as follows. If there is a previously stored LSN in the Kafka offsets topic, the connector continues streaming changes from that position. If no LSN has been stored, the connector starts streaming changes from the point in time when the PostgreSQL logical replication slot was created on the server. This snapshot mode is useful only when you know all data of interest is still reflected in the WAL.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3279]:
```suggestion
`no_data`:: The connector never performs snapshots. 
When a connector is configured this way, after it starts, it behaves as follows:

If there is a previously stored LSN in the Kafka offsets topic, the connector continues streaming changes from that position.
If no LSN is stored, the connector starts streaming changes from the point in time when the PostgreSQL logical replication slot was created on the server. 
Use this snapshot mode only when you know all data of interest is still reflected in the WAL.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 16]
```diff
@@ -3418,31 +3267,33 @@ For example, +
...
+
+`never`:: Deprecated see `no_data`.
+
+`when_needed`:: The connector performs a snapshot upon startup whenever it deems it necessary. That is, when no offsets are available, or when a previously recorded offset specifies a log position that is not available in the server.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3283]:
```suggestion
`when_needed`:: After the connector starts, it performs a snapshot only if it detects one of the following circumstances:
 
* It cannot detect any topic offsets.
* A previously recorded offset specifies a log position that is not available on the server.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 17]
```diff
@@ -3418,31 +3267,33 @@ For example, +
...
+
 For more information, see the xref:#postgresql-connector-snapshot-mode-options[table of `snapshot.mode` options].
 
 ifdef::community[]
 |[[postgresql-property-snapshot-mode-custom-name]]<<postgresql-property-snapshot-mode-custom-name, `+snapshot.mode.custom.name+`>>
 |No default
 | When `snapshot.mode` is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.Snapshotter' interface.
 The provided implementation is called after a connector restart to determine whether to perform a snapshot.
-For more information, see xref:postgresql-custom-snapshot[custom snapshotter SPI].
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3296]: Similar comment about ensuring unique IDs if content becomes productized.

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 18]
```diff
@@ -3468,15 +3319,15 @@ ifdef::community[]
 |[[postgresql-property-snapshot-locking-mode-custom-name]]<<postgresql-property-snapshot-locking-mode-custom-name, `+snapshot.locking.mode.custom.name+`>>
 |No default
 | When `snapshot.locking.mode` is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotLock' interface.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3321]:
```suggestion
| When xref:postgresql-property-snapshot-locking-mode[`snapshot.locking.mode`] is set to `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotLock' interface.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 19]
```diff
@@ -3468,15 +3319,15 @@ ifdef::community[]
...
-For more information, see xref:postgresql-custom-snapshot[custom snapshotter SPI].
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3322]: Ditto.

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 20]
```diff
@@ -3488,7 +3339,7 @@ ifdef::community[]
 |[[postgresql-property-snapshot-snapshot-query-mode-custom-name]]<<postgresql-property-snapshot-query-mode-custom-name, `+snapshot.query.mode.custom.name+`>>
 |No default
 | When xref:postgresql-property-snapshot-query-mode[`snapshot.query.mode`] is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotQuery' interface.
-For more information, see xref:postgresql-custom-snapshot[custom snapshotter SPI].
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3342]: Ditto.

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 21]
```diff
@@ -0,0 +1,149 @@
+For more advanced uses, you can fine-tune control the snapshot by implementing one of the following interfaces:
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:1]:
```suggestion
For more advanced uses, you can fine-tune control of the snapshot by implementing one of the following interfaces:
```

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+ * - Whether the snapshot includes schema (if supported). <br>
+ * - Whether to snapshot data or schema following an error.
+ * <p>
+ * Although Debezium provides many default snapshot modes,
+ * to provide more advanced functionality, such as partial snapshots,
+ * you can customize implementation of the interface.
+ * For more information, see the documentation.
+ *
+ *
+ * @author Mario Fiore Vitale
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:25]: I think that I missed this before, but is there any reason why you wouldn't want your name in the productized documentation?

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+}
+----
+
+[source,java,indent=0,subs="+attributes"]
+.io.debezium.snapshot.spi.SnapshotQuery interface. All built-in snapshot query modes implement this interface.
+----
+/**
+ * {@link SnapshotQuery} is used to determine the query used during a data snapshot
+ *
+ * @author Mario Fiore Vitale
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:100]: Redact name?

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 24]
```diff
@@ -0,0 +1,149 @@
...
+}
+----
+
+[source,java,indent=0,subs="+attributes"]
+.io.debezium.snapshot.spi.SnapshotLock interface. All built-in snapshot lock modes implement this interface.
+----
+/**
+ * {@link SnapshotLock} is used to determine the table lock mode used during schema snapshot
+ *
+ * @author Mario Fiore Vitale
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:131]: Redact name?

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 25]
```diff
@@ -630,6 +630,56 @@ a|Record the successful completion of the snapshot in the connector offsets.
...
+WARNING: Do not use this mode to perform a snapshot if schema changes were committed to the database after the last connector shutdown.
+
+|`when_needed`
+|The connector performs a snapshot upon startup whenever it deems it necessary. That is, when no offsets are available, or when a previously recorded offset specifies a log position that is not available in the server.
+
+ifdef::community[]
+|`custom`
+|The `custom` snapshot mode lets you inject your own implementation of the `io.debezium.spi.snapshot.Snapshotter` interface.
+Set the `snapshot.mode.custom.name` configuration property to the name provided by the `name()` method of your implementation, as specified on the classpath of your Kafka Connect cluster, or included in the connector JAR file, if using the `EmbeddedEngine`.
+For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
```
[the remark this answers is not in the document]
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:676]: @jpechane The Snapshotter interface was marked for community only for Postgres connector, should we maintain in this way also now that is available for the other connectors?

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 26]
```diff
@@ -3347,32 +3403,74 @@ For example, if you set `max.queue.size=1000`, and `max.queue.size.in.bytes=5000
...
+
+`minimal_percona`:: the connector holds link:https://www.percona.com/doc/percona-server/8.0/management/backup_locks.html[the global backup lock] for only the initial portion of the snapshot during which the connector reads the database schemas and other metadata. The remaining work in a snapshot involves selecting all rows from each table. The connector can do this in a consistent fashion by using a REPEATABLE READ transaction. This is the case even when the global backup lock is no longer held and other MySQL clients are updating the database. This mode does not flush tables to disk, is not blocked by long-running reads, and is available only in Percona Server. +
+
+`extended`:: blocks all writes for the duration of the snapshot. Use this setting if there are clients that are submitting operations that MySQL excludes from REPEATABLE READ semantics. +
```
[the remark this answers is not in the document]
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:3440]: I think it's better now ;-)

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 27]
```diff
@@ -0,0 +1,149 @@
...
```
Comment (mfvitale) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:131]:
```suggestion
 
```

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 28]
```diff
@@ -0,0 +1,149 @@
...
+}
+----
+
+[source,java,indent=0,subs="+attributes"]
+.io.debezium.snapshot.spi.SnapshotQuery interface. All built-in snapshot query modes implement this interface.
+----
+/**
+ * {@link SnapshotQuery} is used to determine the query used during a data snapshot
+ *
+ * @author Mario Fiore Vitale
```
Comment (mfvitale) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:100]:
```suggestion
 * 
```

File: documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc [hunk 29]
```diff
@@ -0,0 +1,149 @@
...
+ * - Whether the snapshot includes schema (if supported). <br>
+ * - Whether to snapshot data or schema following an error.
+ * <p>
+ * Although Debezium provides many default snapshot modes,
+ * to provide more advanced functionality, such as partial snapshots,
+ * you can customize implementation of the interface.
+ * For more information, see the documentation.
+ *
+ *
+ * @author Mario Fiore Vitale
```
Comment (mfvitale) [inline documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc:25]:
```suggestion
 *
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 30]
```diff
@@ -3418,31 +3277,41 @@ For example, +
...
-`always` - The connector performs a snapshot each time the connector starts. +
- +
-`never` - The connector never performs snapshots. When a connector is configured this way, its behavior when it starts is as follows. If there is a previously stored LSN in the Kafka offsets topic, the connector continues streaming changes from that position. If no LSN has been stored, the connector starts streaming changes from the point in time when the PostgreSQL logical replication slot was created on the server. The `never` snapshot mode is useful only when you know all data of interest is still reflected in the WAL. +
- +
-`initial_only` - The connector performs an initial snapshot and then stops, without processing any subsequent changes. +
- +
-`exported` -  deprecated +
- +
+
+`always`:: The connector performs a snapshot each time the connector starts.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3283]:
```suggestion
`always`:: The connector performs a snapshot every time that it starts.
The snapshot includes the structure and data of the captured tables.
Specify this value to populate topics with a complete representation of the data from the captured tables every time that the connector starts.
After the snapshot completes, the connector begins to stream event records for subsequent database changes.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 31]
```diff
@@ -630,6 +630,62 @@ a|Record the successful completion of the snapshot in the connector offsets.
...
 |===
 
+[id="mysql-connector-snapshot-mode-options"]
+.Settings for `snapshot.mode` connector configuration property
+[cols="30%a,70%a",options="header"]
+|===
+|Setting |Description
+
+|`always`
+|The connector performs a database snapshot after every restart.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:640]:
```suggestion
|The connector performs a snapshot every time that it starts.
The snapshot includes the structure and data of the captured tables.
Specify this value to populate topics with a complete representation of the data from the captured tables every time that the connector starts.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 32]
```diff
@@ -3347,32 +3409,97 @@ For example, if you set `max.queue.size=1000`, and `max.queue.size.in.bytes=5000
 |Positive integer value that specifies the number of milliseconds the connector should wait for new change events to appear before it starts processing a batch of events. Defaults to 500 milliseconds, or 0.5 second.
 
 |[[mysql-property-snapshot-mode]]<<mysql-property-snapshot-mode, `+snapshot.mode+`>>
-|`initial`
+|_initial_
 |Specifies the criteria for running a snapshot when the connector starts. Possible settings are: +
+
+`always`:: The snapshot includes the structure and data of the captured tables.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3415]:
```suggestion
`always`:: The connector performs a snapshot every time that it starts.
The snapshot includes the structure and data of the captured tables.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
Several more changes for consistency. 
As stated in one comment, for now, let's drop the issue that I raised about unique IDs. We can revisit that at a later point if we confirm that we'll expose the content downstream.

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 33]
```diff
@@ -3347,32 +3409,97 @@ For example, if you set `max.queue.size=1000`, and `max.queue.size.in.bytes=5000
...
+Specify this value to populate topics with a complete representation of the data from the captured tables every time that the connector starts.
+
+`initial`:: The connector runs a snapshot only when no offsets have been recorded for the logical server name.
```
[the remark this answers is not in the document]
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:3418]: I would add that  it runs also when a not completed snapshot is found.

File: documentation/modules/ROOT/pages/connectors/mysql.adoc (hunk 25 above)
[the remark this answers is not in the document]
Comment (jpechane) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:685]: @mfvitale Let's keep it community only now. We'll decide if it is ready for productization base od 2.6 community feedback.

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 34]
```diff
@@ -3347,32 +3409,97 @@ For example, if you set `max.queue.size=1000`, and `max.queue.size.in.bytes=5000
 |Positive integer value that specifies the number of milliseconds the connector should wait for new change events to appear before it starts processing a batch of events. Defaults to 500 milliseconds, or 0.5 second.
 
 |[[mysql-property-snapshot-mode]]<<mysql-property-snapshot-mode, `+snapshot.mode+`>>
-|`initial`
+|_initial_
```
[the remark this answers is not in the document]
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:3415]: I found the same inconsistency and I don't have strong opinion on what would be better. Feel free to choose what you think is the best.

[the remark this answers is not in the document]
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/mysql.adoc:3415]: For expediency, let's keep your version here for now. If I find that we have a standard, or if we decide we should implement one, we can do a full-scale sweep through all of the docs to apply consistent formatting.

**Comment by mfvitale** [comment]
@roldanbob do you think are we ready to merge?

**Pull request merged**